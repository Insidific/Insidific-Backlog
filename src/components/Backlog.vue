<template>
<div class="backlog">
  <div v-for="(game, gameKey) in backLog" :key="gameKey">
    <BacklogItem :game="game" />
  </div>
</div>
</template>

<script>
import BacklogItem from "./BacklogItem.vue";
export default {
  name: "Backlog",
  components: {
    BacklogItem
  },
  props: ['backLog']
};
</script>

<style scoped lang="scss">
.backlog {
  padding-left: 25px;
  padding-right: 25px;
}
</style>
