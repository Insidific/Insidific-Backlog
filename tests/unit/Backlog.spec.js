import { mount } from "@vue/test-utils";
import Vue from "vue";
import Backlog from "@/components/Backlog.vue";
import VueMaterial from "vue-material";

describe("Backlog.vue", () => {
  beforeAll(() => {
    Vue.use(VueMaterial);
  });

  it("Backlog to Match Snapshot", () => {
    const games = {
      Game1: {
        name: "Marvel's Spider-Man",
        developer: "Insomniac Games",
        description:
          "Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.",
        image:
          "https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg",
        completion: "incomplete"
      }
    };
    const wrapper = mount(Backlog, { propsData: { backLog: games } });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Backlog to Contain Backlog items, title, image, developer and description", () => {
    const games = {
      Game1: {
        name: "Marvel's Spider-Man",
        developer: "Insomniac Games",
        description:
          "Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.",
        image:
          "https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg",
        completion: "incomplete"
      }
    };
    const wrapper = mount(Backlog, { propsData: { backLog: games } });

    expect(wrapper.html()).toContain("<h1>Marvel's Spider-Man</h1>");
    expect(wrapper.html()).toContain('<img src="https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg" alt="Cover">');
    expect(wrapper.html()).toContain("<p>Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.</p>");
    expect(wrapper.html()).toContain("<h3>Insomniac Games</h3>");
  });

  it("BacklogItem should have class with Complete", () => {
    const games = {
      Game1: {
        name: "Marvel's Spider-Man",
        developer: "Insomniac Games",
        description:
          "Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.",
        image:
          "https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg",
        completion: "incomplete"
      }
    };
    const wrapper = mount(Backlog, { propsData: { backLog: games } });

    expect(wrapper.contains(".incomplete")).toBe(true);
  });

  it("BacklogItem should have class with Complete", () => {
    const games = {
      Game1: {
        name: "Marvel's Spider-Man",
        developer: "Insomniac Games",
        description:
          "Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.",
        image:
          "https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg",
        completion: "complete"
      }
    };
    const wrapper = mount(Backlog, { propsData: { backLog: games } });

    expect(wrapper.contains(".complete")).toBe(true);
  });
})