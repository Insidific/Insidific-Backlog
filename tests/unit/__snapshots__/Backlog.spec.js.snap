// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backlog.vue Backlog to Match Snapshot 1`] = `
<div class="backlog">
  <div>
    <div class="backlog-item">
      <div class="md-card md-theme-default incomplete">
        <div class="md-card-media"><img src="https://images.igdb.com/igdb/image/upload/t_cover_big/xcrfutqlaa3q81m7blmh.jpg" alt="Cover"></div>
        <div class="md-card-content">
          <h1>Marvel's Spider-Man</h1>
          <h3>Insomniac Games</h3>
          <p>Starring the world’s most iconic Super Hero, Spider-Man features the acrobatic abilities, improvisation and web-slinging that the wall-crawler is famous for, while also introducing elements never-before-seen in a Spider-Man game. From traversing with parkour and utilizing the environment, to new combat and blockbuster set pieces, it’s Spider-Man unlike any you’ve played before.</p>
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
